### DISCUSSION TITLE
Option for Field type address to have Zip before City

### DISCUSSION THREAD

**Discussion by Megafry** [verbb/formie#1678] [Ideas] [answered]
For addresses in Europe, the zip code is placed before the city.
Therefore it would be handy to have the option to change the order in the field settings.

I was able to change it with the event `EVENT_MODIFY_FRONT_END_SUBFIELDS`.
but a native solution would be welcome.
```php
Event::on(Address::class, Address::EVENT_MODIFY_FRONT_END_SUBFIELDS, function(ModifyFrontEndSubfieldsEvent $event) {
    //move the field zip before the city
    foreach ($event->rows as $key => $rowConfig) {
        foreach ($rowConfig as $config) {
            $handle = $config['handle'];
            if ($handle == 'zip') {
                $event->rows [$key] = array_reverse($rowConfig);
                break 2;
            }
        }
    }
});
```
best regards

**Comment by engram-design** [comment, marked as answer]
Totally understand that's the norm for Europe, but I'm pretty sure that's not for Australia, the US and a few other places and makes much more logical sense to put next to country and state (after city). But in that instance, it'll be addressed in https://github.com/verbb/formie/issues/490 so that you'll be able to re-order these fields as required.

**Reply by Megafry** [reply]
Thanks for the quick answer, nice to know this will be added in the next major update.

**Comment by engram-design** [comment]
Added in [3.0.0](https://github.com/verbb/formie/releases/tag/3.0.0)
